src/config.rs

```diff
@@ -375,6 +376,29 @@ impl<'a> Cfg<'a> {
         self.toolchain_override = Some(toolchain_override.to_owned());
     }
 
+    pub(crate) fn set_auto_install(&mut self, mode: AutoInstallMode) -> Result<()> {
+        self.settings_file.with_mut(|s| {
+            s.auto_install = Some(mode);
+            Ok(())
+        })?;
+        (self.notify_handler)(Notification::SetAutoInstall(mode.as_str()));
+        Ok(())
+    }
+
+    pub(crate) fn should_auto_install(&self) -> Result<bool> {
+        // If auto install is disabled in settings then the environment variable
```

Comment:
Just to be sure, is this coherent with what we have for now? Somehow I'd expect the env var to have the higher precedence here...

https://github.com/rust-lang/rustup/blob/35d97428d259fcc3dabf1dd1c10b65ae5c0d0be2/doc/user-guide/src/overrides.md?plain=1#L7-L12

Reply:
Yeah, I wasn't sure about that. My thought was to make it act like 1.28.0 (where the environment var didn't exist). But rethinking it, I've changed my mind and now I think the environment var should take precedence.

Reply:
Ok, done.